### PR TITLE
SEP 054: package installation and lookup

### PR DESCRIPTION
**Pull request by jakebeal**
This code is still pretty preliminary and architecturally messy, but I want to give a change for folks to look it over.

### CODE REVIEW

**Comment by hgscott** [comment]
Less of a question about packages and more just a question about namespaces:
When you define a package from BBa_J23101.nt, the namespace defining the package is “https://synbiohub.org”, which is correct from the code since all of the objects in the file have that as the value for “hasNamespace”. 
But the first URI for each line for each object (the subjects?) all start with “https://synbiohub.org/public/igem/”, why is that not the namespace? And should that be the name of the package?

**Comment by jakebeal** [comment]
There is flexibility in that question: one could do this in a number of different ways. 
If you don't supply a namespace, however, as was done in creating the test case, then it guesses on by default, and it looks like the sbol2to3 conversion is defaulting to the shortest possible namespace, which is likely the wrong default. I've opened an issue to use the default from pySBOL3 instead: #156

**Review by hgscott — APPROVED** [review]
Ok, thanks for the explanation, verything else made sense to me.

File: test/test_package.py [hunk 1]
```diff
@@ -210,6 +209,43 @@ def test_make_package_from_MyPackage_w_sub_sub_packages(self):
...
+
+    def test_cross_document_lookup(self):
+        """Test that package system correctly overrides the document lookup function to enable cross-package lookups"""
+        # What I want for a load pattern:
+        # sip install root-package-dir
+        # Materials will be stored in Path.home() / ".sip"
+        # A catalog is stored in Path.home() / ".sip" / "installed-packages.nt"
+        # iGEM materials are stored in Path.home() / ".sip" / "igem"
+        # sbol_utilities.package.load_package('igem')
+        package.load_package('https://synbiohub.org/public/igem', TEST_FILES / 'BBa_J23101.nt')
```
[the remark this answers is not in the document]
Comment (jakebeal) [reply test/test_package.py:240]: Good  catch! That should be an error, I think. I will fix that.

**Review by tcmitchell — APPROVED** [review]
This all looks fine to me. I don't fully understand how this is intended to work for a user. It might be nice to have a bit of a usage guide at some point. It's probably too early for that though. Something to think about for the future.

File: test/test_package.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakebeal) [reply test/test_package.py]: I've added validation, also re-discovering that we intended to have the Package object itself be in the package build artifact.

**Pull request merged**